test/TestDestroy.cpp

```diff
@@ -27,17 +27,27 @@ TEST(TestDestroy, TestDestroyTensorSingle)
         {
             kp::Manager mgr;
 
-            tensorA = mgr.tensor({ 0, 0, 0 });
+            const std::vector<float> initialValues = {0.0f, 0.0f, 0.0f};
+
+            tensorA = mgr.tensor(initialValues);
 
             std::shared_ptr<kp::Algorithm> algo =
-              mgr.algorithm({ tensorA }, spirv);
+              mgr.algorithm({tensorA}, spirv);
+
+            // Sync values to and from device
+            mgr.sequence()
+            ->eval<kp::OpTensorSyncDevice>(algo->getTensors())
+            ->eval<kp::OpTensorSyncLocal>(algo->getTensors());
```

Comment:
Here you don't really need SyncLocal - this would be executed at the end to retrieve the results